File: src/test/java/net/mreunionlabs/tapgradle/generate/generator/JavaStructureGeneratorTest.java
package net.mreunionlabs.tapgradle.generate.generator;

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

/**
 * Created by abangkis on 6/16/2016.
 */
public class JavaStructureGeneratorTest {
    private GeneratePluginExtension ext;

    @BeforeMethod
    public void setUp() throws Exception {
        ext = new GeneratePluginExtension();
        ext.setPackageString("net.mreunionlabs.test");
        ext.setJavaDir(".");
    }

    @Test
    public void testCreateDirectories() throws Exception {
        JavaStructureGenerator generator = new JavaStructureGenerator();
        generator.createDirectories(ext);
    }

}

File: src/test/kotlin/net/mreunionlabs/tapgradle/generate/structure/JavaStructureGeneratorTest.kt
package net.mreunionlabs.tapgradle.generate.structure

import io.kotlintest.TestCase
import io.kotlintest.specs.StringSpec
import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension
import org.gradle.api.Project
import org.gradle.testfixtures.ProjectBuilder

class JavaStructureGeneratorTest : StringSpec() {
    private lateinit var project: Project
    private lateinit var ext: GeneratePluginExtension

    override fun beforeTest(testCase: TestCase) {
        project = ProjectBuilder.builder().build()
        ext = GeneratePluginExtension()
        ext.packageString = "net.mreunionlabs.test"
        ext.javaDir = "gen/java"
    }

    init {
        "create appmodule test should run" {
            JavaStructureGenerator().createDirectories(ext)
        }
    }
}